scipy/io/mmio.py

```diff
@@ -739,22 +748,21 @@ def _write(self, stream, a, comment='', field=None, precision=None,
             # write shape spec
             stream.write(asbytes('%i %i %i\n' % (rows, cols, coo.nnz)))
 
-            # make indices and data array
+            template = self._field_template(field, precision-1)
+
             if field == self.FIELD_PATTERN:
-                IJV = vstack((coo.row, coo.col)).T
-            elif field in [self.FIELD_INTEGER, self.FIELD_REAL]:
-                IJV = vstack((coo.row, coo.col, coo.data)).T
+                for r, c in zip(coo.row+1, coo.col+1):
+                    stream.write(asbytes("%i %i\n" % (r, c)))
+            elif field in (self.FIELD_INTEGER, self.FIELD_REAL):
+                for r, c, d in zip(coo.row+1, coo.col+1, coo.data):
+                    stream.write(asbytes(("%i %i " % (r, c)) +
+                                         (template % d)))
             elif field == self.FIELD_COMPLEX:
-                IJV = vstack((coo.row, coo.col, coo.data.real,
-                              coo.data.imag)).T
+                for r, c, d in zip(coo.row+1, coo.col+1, coo.data):
+                    stream.write(asbytes(("%i %i " % (r, c)) +
+                                         (template % (d.real, d.imag))))
             else:
                 raise TypeError('Unknown field type %s' % field)
-            IJV[:, :2] += 1  # change base 0 -> base 1
-
-            # formats for row indices, col indices and data columns
-            fmt = ('%i', '%i') + ('%%.%dg' % precision,) * (IJV.shape[1]-2)
-            # save to file
-            savetxt(stream, IJV, fmt=fmt)
 
 
 def _is_fromfile_compatible(stream):
```

Comment:
I haven't really measured the performance impact. I'll try to do that later today. I guess my question of removing goes into the direction of keeping it for the case that fromfile is fixed. One thing to note though is, that dense matrices and sparse matrices if the corresponding scipy type is not available are all read by use of a python loop as well. (see lines 502-564)